### PR TITLE
Remove spring test modules

### PR DESCRIPTION
**Pull request by moarychan**
The new features are supported in the 4.0 feature, and many modules have been refactored, 
so we plan to remove the `azure-spring-boot-test-xxx` modules to make the spring project structure cleaner, 
more unit tests should be included, some unit tests have been added in the previous 
PR https://github.com/Azure/azure-sdk-for-java/pull/25820, others will keep or remove.

Changelog:
- Remove `azure-spring-boot-test-application`
- Remove `azure-spring-boot-test-core`
- Remove `azure-spring-boot-test-cosmos`
- Remove `azure-spring-boot-test-keyvault`
- Remove `azure-spring-boot-test-parent`
- Remove `azure-spring-boot-test-servicebus-jms`
- Remove `azure-spring-boot-test-storage`
- Remove `spring-cloud-azure-stream-binder-test`
- Remove `spring-cloud-azure-test-parent`
- Remove `azure-spring-boot-test-keyvault-certificate`

### CODE REVIEW

**Comment by moarychan** [comment]
@saragluna , please help review, thanks!

File: sdk/spring/spring-cloud-azure-stream-binder-servicebus/pom.xml [hunk 1]
```diff
@@ -114,6 +113,22 @@
           </rules>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-servicebus/pom.xml:117]: Why is this being added?

File: sdk/spring/spring-cloud-azure-test-eventhubs-binder/pom.xml [hunk 2]
```diff
@@ -31,6 +31,27 @@
...
+        <version>2021.0.0</version> <!-- {x-version-update;org.springframework.cloud:spring-cloud-dependencies;external_dependency} -->
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-core</artifactId>
+        <version>1.23.1</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+      </dependency>
+    </dependencies>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-test-eventhubs-binder/pom.xml:52]: do we still need this?

File: sdk/spring/spring-cloud-azure-test-servicebus-binder/pom.xml [hunk 3]
```diff
@@ -31,6 +31,27 @@
...
+        <artifactId>spring-cloud-dependencies</artifactId>
+        <version>2021.0.0</version> <!-- {x-version-update;org.springframework.cloud:spring-cloud-dependencies;external_dependency} -->
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-core</artifactId>
+        <version>1.23.1</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+      </dependency>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-test-servicebus-binder/pom.xml:51]: same here

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-stream-binder-servicebus/pom.xml (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-stream-binder-servicebus/pom.xml]: 😓

File: sdk/spring/spring-cloud-azure-test-eventhubs-binder/pom.xml (hunk 2 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-test-eventhubs-binder/pom.xml]: No, let me remove it.

**Review by saragluna — APPROVED** [review]
LGTM, thanks for the change!

**Pull request merged**